### ISSUE TITLE
Clicking off a dialog should not count as closing/cancelling it

### ISSUE THREAD

**Issue by LarsKemmann** [CareTogether/CareTogetherCMS#280]
This change needs to be applied consistently to every dialog in the client-side codebase. Several dialogs use the `<UpdateDialog />` custom component, so that only requires a single change; other dialogs use the MUI `<Dialog />` component directly and will need to be updated individually.

Guidance can be found in [the MUI documentation for `<Dialog />`](https://mui.com/material-ui/api/dialog/). Essentially, the `onClose` callback needs to filter the `reason` and only perform the close action if `reason !== "backdropClick"`.

**Comment by LarsKemmann** [comment]
To do this right, the Material-UI V5 upgrade should happen first.

**Comment by LarsKemmann** [comment]
This is now ready to be worked on.

**Comment by donschaefer** [comment]
Not diving into this just yet (as it seems it'll likely involve a good bit of individual changes & testing), but wanted to share a link that may provide some value to whoever ends up working on it (even if that ends up being me eventually)...

https://smartdevpreneur.com/how-to-disable-mui-dialog-backdrop-click/

**Comment by LarsKemmann** [comment]
#654 addressed most of this. When I tested it, some dialogs still closed in response to clicking their backdrop (adding a new volunteer family, for example), so this issue needs another PR to fully resolve it.
